src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs

```diff
@@ -9,36 +9,41 @@
 namespace D2L.Security.OAuth2.Keys.Default.Data {
 	internal sealed class JwksProvider : IJwksProvider {
 		private readonly HttpClient m_httpClient;
-		private readonly Uri m_authEndpoint;
+		private readonly Uri m_jwksEndpoint;
+		private readonly Uri m_jwkEndpoint;
 
 		public JwksProvider(
 			HttpClient httpClient,
-			Uri authEndpoint
+			Uri jwksEndpoint,
+			Uri jwkEndpoint
 		) {
 			m_httpClient = httpClient;
-			m_authEndpoint = authEndpoint;
+			m_jwksEndpoint = jwksEndpoint;
+			m_jwkEndpoint = jwkEndpoint;
 		}
 
 		async Task<JsonWebKeySet> IJwksProvider.RequestJwksAsync() {
-			var url = GetJwksEndpoint( m_authEndpoint );
-
 			try {
-				using( HttpResponseMessage response = await m_httpClient.GetAsync( url ).SafeAsync() ) {
+				using( HttpResponseMessage response = await m_httpClient.GetAsync( m_jwksEndpoint ).SafeAsync() ) {
 					response.EnsureSuccessStatusCode();
 					string jsonResponse = await response.Content.ReadAsStringAsync().SafeAsync();
-					var jwks = new JsonWebKeySet( jsonResponse, url );
+					var jwks = new JsonWebKeySet( jsonResponse, m_jwksEndpoint );
 					return jwks;
 				}
 			} catch( HttpRequestException e ) {
-				throw CreateException( e, url );
+				throw CreateException( e, m_jwksEndpoint );
 			} catch( JsonWebKeyParseException e ) {
-				throw CreateException( e, url );
+				throw CreateException( e, m_jwksEndpoint );
 			}
 		}
 
 		async Task<JsonWebKeySet> IJwksProvider.RequestJwkAsync( string keyId ) {
-			var url = GetJwkEndpoint( m_authEndpoint, keyId );
+			var url = GetJwkEndpoint( m_jwkEndpoint, keyId );
 			try {
+				if( url == null ) {
+					return await ( this as IJwksProvider ).RequestJwksAsync().SafeAsync();
+				}

```

Comment:
Does this need to be in the `try`? It looks like that method catches the same exceptions.

Reply:
Probably not, no.

Reply:
- [X] move block out of try-catch